lib/shopify_api/resources/base.rb

```diff
@@ -4,6 +4,7 @@ module ShopifyAPI
   class Base < ActiveResource::Base
     class InvalidSessionError < StandardError; end
     extend Countable
+    self.format = :json
```

Comment:
If you are using the gem then you dont care what the return format is because AR handles the serialization and deserialization of the request/response. All of shopify's api's support JSON so we might as well only request in JSON.

The reason it fixes things is because in active resource 3.0 it seems like the default format was xml and therefore some http mocks weren't working